### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by odelalleau**
## Motivation

Improve documentation

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

I'm hoping CI will run tests on docs!

## Related Issues and PRs

N/A

### CODE REVIEW

File: website/docs/plugins/submitit_launcher.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ For more details, including descriptions for each parameter, check out the [conf
 
 An [example application](https://github.com/facebookresearch/hydra/tree/master/plugins/hydra_submitit_launcher/example) using this launcher is provided in the plugin repository.
 
-Starting the app with `python my_app.py task=1,2,3 -m` will launch 3 executions (you can override the launcher to run locally for testing by adding `hydra/launcher=submitit_local`):
+Starting the app with `python my_app.py task=1,2,3 -m` (where `-m` is a shortcut for `--multirun`) will launch 3 executions (you can override the launcher to run locally for testing by adding `hydra/launcher=submitit_local`):
```
Comment (omry) [inline website/docs/plugins/submitit_launcher.md:86]: just specify --multirun instead.

File: website/docs/tutorials/basic/running_your_app/3_working_directory.md [hunk 2]
```diff
@@ -81,5 +81,12 @@ to_absolute_path('foo')    : /Users/omry/dev/hydra/foo
...
 
-
 Working directory can be [customized](/configure_hydra/workdir.md).
+
+### Working directory and interpolation
+
+You may want a config option to depend on Hydra's automatically created working directory.
+This can be achieved through interpolation:
+```yaml title="config.yaml"
+logdir: ${hydra:run.dir}/logs
```
Comment (omry) [inline website/docs/tutorials/basic/running_your_app/3_working_directory.md:91]: hmm, this is not good. it will not work if you are doing a sweep.
you should be using logs relative to cwd.

File: website/docs/tutorials/structured_config/1_minimal_example.md [hunk 3]
```diff
@@ -83,5 +83,5 @@ It can be useful when you care about the methods or attributes of an object, not
 ## Overriding default values in the `@dataclass`
 You can use instances of the dataclasses to override default values in the stored config.
 ```python
-cs.store(name="config", node=MySQLConfig(user="root", password="1234"))
+cs.store(name="config", node=MySQLConfig(host="anotherhost", port=80))
```
Comment (omry) [inline website/docs/tutorials/structured_config/1_minimal_example.md:86]: why this change?

File: website/docs/tutorials/structured_config/3_config_groups.md [hunk 4]
```diff
@@ -48,6 +48,9 @@ db:
   user: postgre_user
 ```
 
+In the above example we need the `+` because there is no default choice for which config to load in the `db` group, so we are effectively _adding_ a new config element rather than _overriding_ an existing one.
+The next page in this tutorial will show how to add such a default choice, thus removing the need for `+` on the command line.
```
Comment (omry) [inline website/docs/tutorials/structured_config/3_config_groups.md:52]:
```suggestion
The `+` above is required because there is no default choice for the config group `db`.
The next page will re-introduce the Defaults List, eliminating the need for the `+`.
```

File: website/docs/tutorials/structured_config/4_defaults.md [hunk 5]
```diff
@@ -3,11 +3,11 @@ id: defaults
 title: Defaults
 ---
 
-You can define a defaults list in your primary config just like you can in your primary config file.
-The example below extends the previous example by adding a defaults list that will load db=mysql by default.
+You can define a defaults list in your primary (structured) config just like you can in your primary (YAML) config file.
+The example below extends the previous example by adding a defaults list that will load `db=mysql` by default.
```
Comment (omry) [inline website/docs/tutorials/structured_config/4_defaults.md:7]:
```suggestion
You can define a defaults list in your primary Structured Config just like you can in your primary `config.yaml` file.
The example below extends the previous example by adding a defaults list that will load `db=mysql` by default.
```

File: website/docs/tutorials/structured_config/5_schema.md [hunk 6]
```diff
@@ -2,11 +2,11 @@
 id: schema
 title: Structured config schema
 ---
-We have seen how to use Structured Configs as configuration, but they can also be used as a schema validating configuration files!  
-When Hydra loads a configuration, it looks for config with the same name in the `ConfigStore`.
+We have seen how to use Structured Configs as configuration, but they can also be used as schema validating configuration files!
```
Comment (omry) [inline website/docs/tutorials/structured_config/5_schema.md:5]: disagreeing, the a makes sense to me.

File: website/docs/tutorials/structured_config/5_schema.md [hunk 7]
```diff
@@ -2,11 +2,11 @@
...
+When Hydra loads a configuration file, it looks for a config with the same name in the `ConfigStore`.
```
Comment (omry) [inline website/docs/tutorials/structured_config/5_schema.md:6]: hmm, not accurate.
the configuration can be provided by other config sources that are not file based.

File: website/docs/tutorials/structured_config/5_schema.md [hunk 8]
```diff
@@ -40,20 +40,20 @@ class MySQLConfig(DBConfig):
...
     timeout: int = 10
 
 
 # registering db/mysql and db/postgresql schemas.
-ss = ConfigStore.instance()
-ss.store(group="db", name="mysql", node=MySQLConfig)
-ss.store(group="db", name="postgresql", node=PostGreSQLConfig)
+cs = ConfigStore.instance()
+cs.store(group="db", name="mysql", node=MySQLConfig)
+cs.store(group="db", name="postgresql", node=PostGreSQLConfig)
```
Comment (omry) [inline website/docs/tutorials/structured_config/5_schema.md:52]: yikes, as you can tell - documentations are not tested automatically.

File: website/docs/tutorials/structured_config/5_schema.md [hunk 9]
```diff
@@ -40,20 +40,20 @@ class MySQLConfig(DBConfig):
...
 
 
 # config here is config.yaml under the conf directory.
-# config.yaml will compose in db: mysql by default (per the defaults list),
+# config.yaml will compose in database: mysql by default (per the defaults list),
```
Comment (omry) [inline website/docs/tutorials/structured_config/5_schema.md:56]: should be db. if this is wrong we should fix database to db, not the other way around.

File: website/docs/plugins/submitit_launcher.md (hunk 1 above)
Comment (odelalleau) [reply website/docs/plugins/submitit_launcher.md:86]: That was my first thought but then I realized that if the doc only showed `--multirun` then I'd still be using that myself and wasting time typing all these extra characters.
What about:
```
Starting the app with `python my_app.py task=1,2,3 --multirun` (`-m` in abbreviated form) ...
```

File: website/docs/tutorials/structured_config/1_minimal_example.md (hunk 3 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/1_minimal_example.md:86]: This example has no `user` / `password` fields => they trigger an error

File: website/docs/tutorials/structured_config/3_config_groups.md (hunk 4 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/3_config_groups.md]: I took your changes

File: website/docs/tutorials/structured_config/4_defaults.md (hunk 5 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/4_defaults.md]: Took your changes

File: website/docs/tutorials/structured_config/5_schema.md (hunk 7 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/5_schema.md:6]: Ok will revert the "file" part, however I'm curious how else you can provide a config? The two ways I'm aware of are (1) through a file, or (2) through the config store, but here (2) wouldn't make sense, would it?

File: website/docs/tutorials/structured_config/5_schema.md (hunk 9 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/5_schema.md:56]: Ah right I didn't pay enough attention to the diff as copy / pasted from the code ==> will fix the code and docs in another commit

File: website/docs/tutorials/structured_config/5_schema.md (hunk 6 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/5_schema.md:5]: "a" is singular so doesn't play well with "files" IMO -- I actually just asked an American colleague who confirmed (also suggested to rpelace "as configuration" by "for configuration", so I'll add another commit about that)

Comment (odelalleau) [reply website/docs/tutorials/structured_config/5_schema.md:5]: Scratch that, I think I just understood the meaning of that sentence... will rephrase in another commit as it's ambiguous

File: website/docs/plugins/submitit_launcher.md (hunk 1 above)
Comment (omry) [reply website/docs/plugins/submitit_launcher.md:86]: You can't explain everything everywhere.
This is described [here](https://hydra.cc/docs/next/tutorials/basic/running_your_app/multi-run).
Maybe add a link if you think it's better.

File: website/docs/tutorials/structured_config/1_minimal_example.md (hunk 3 above)
Comment (omry) [reply website/docs/tutorials/structured_config/1_minimal_example.md:86]: Oh, I see.
```suggestion
cs.store(name="config", node=MySQLConfig(host="test.db", port=3307))
```

File: website/docs/tutorials/structured_config/5_schema.md (hunk 7 above)
Comment (omry) [reply website/docs/tutorials/structured_config/5_schema.md:6]: The config store would not make sense, true.
You can provide it as a file (actual file in the file system) and as a resource (file packaged with your par).
there is an abstraction for ConfigSource that is currently not supporting anything else but I did have a prototype that fetched config files directly from GitHub using some API (it was just too slow to be useful).

You could also imagine configs in some config storage system (configerator), or configs based on a database or some other data storage like LDAP.
the sky is the limit :)

File: website/docs/tutorials/structured_config/5_schema.md (hunk 6 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/5_schema.md]: Let me know if a732a1bcb476e3fdd547cf82e9c7a277d2270089 works for you

File: website/docs/plugins/submitit_launcher.md (hunk 1 above)
Comment (odelalleau) [reply website/docs/plugins/submitit_launcher.md]: What about 39ba5d5e?

File: website/docs/tutorials/structured_config/1_minimal_example.md (hunk 3 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/1_minimal_example.md:86]: Taking your change, but FYI I had intentionally used port 80 because it triggered something "interesting" in the output

File: website/docs/tutorials/structured_config/5_schema.md (hunk 7 above)
Comment (odelalleau) [reply website/docs/tutorials/structured_config/5_schema.md]: Ok thanks => I removed the "file" in 16ef8dad

File: website/docs/tutorials/basic/running_your_app/3_working_directory.md (hunk 2 above)
Comment (odelalleau) [reply website/docs/tutorials/basic/running_your_app/3_working_directory.md:91]: Arg, I didn't realize that the run dir and the working dir were different :/
I'll need to play a bit with all these different folders to get a better hang of how things are working in practice, and how the doc might make this clearer. In the meantime I'll get rid of this commit in the PR.

**Comment by omry** [comment]
Please rebase, the lint problem is resolved in master.

**Comment by odelalleau** [comment]
> Please rebase, the lint problem is resolved in master.

done

**Comment by omry** [comment]
@shagunsodhani yamllint seems to be blowing up suddenly, there were no yaml file changes here.
mind talking a look?

**Comment by shagunsodhani** [comment]
> @shagunsodhani yamllint seems to be blowing up suddenly, there were no yaml file changes here.
> mind talking a look?

I am facing the same problem. https://github.com/facebookresearch/hydra/commit/84feaea52fc29152a7821050ea677101b6345cc9 fixes it but I dont know why it is needed (or only needed on Mac/Win). I could not reproduce this locally on mac (Python 3.8)

More context: https://github.com/facebookresearch/hydra/pull/766#issuecomment-658834787

**Comment by omry** [comment]
please rebase again, the new lint issue is fixed on master.

**Comment by odelalleau** [comment]
> please rebase again, the new lint issue is fixed on master.

done

**Comment by omry** [comment]
Awesome, thanks!
